src/libslic3r/GCode.cpp

```diff
@@ -3791,8 +3792,20 @@ LayerResult GCode::process_layer(
         }
 
         // BBS
+        // int bed_temp = get_bed_temperature(first_extruder_id, false, print.config().curr_bed_type);
+        // gcode += m_writer.set_bed_temperature(bed_temp);
+        // Mark the temperature transition from 1st to 2nd layer to be finished.
+
+        // ELEGOO
+        // If the temperature of the other layer is different from the temperature of the first layer, it needs 
+        // to be applied. To facilitate the setup of a hot bed with multi-zone control in a custom G-CODE.
+        int bed_temp_first = get_bed_temperature(first_printing_extruder_id, true, print.config().curr_bed_type);
```

Comment:
Can you explain the changes in a bit more detail?
Seems the first_printing_extruder_id is not properly assigned anywhere; what are we expecting here?